File: src/1104-Path-In-Zigzag-Labelled-Binary-Tree/1104.cpp
class Solution 
{
public:
    vector<int> pathInZigZagTree(int label) 
    {
        vector<int> res;
        while (true)
        {
            res.emplace_back(label);
            if (label == 1) break;
            label >>= 1;
            label = label ^ ((1 << int(log2(label))) - 1);
        }
        return vector<int>(res.rbegin(), res.rend());
    }
};

File: src/1104-Path-In-Zigzag-Labelled-Binary-Tree/1104.py
class Solution:
    def pathInZigZagTree(self, label: int) -> List[int]:
        res = list()
        while 1:
            res.append(label)
            if label == 1:
                break
            label >>= 1
            label = label ^ (1 << (label.bit_length() - 1)) - 1
        return res[::-1]